lib/instabug_http_client.dart

```diff
@@ -10,11 +10,15 @@ import 'package:instabug_http_client/instabug_http_logger.dart';
 import 'package:meta/meta.dart';
 
 class InstabugHttpClient extends InstabugHttpLogger implements http.Client {
-  InstabugHttpClient() : client = http.Client() {
+  /// Constructs a new [InstabugHttpClient].
+  ///
+  /// Provide a value for [client] in order to override the internal client used
+  /// by this class. This can be useful if you are working with other libraries
+  /// that require other custom client implementations
+  InstabugHttpClient({http.Client? client}) : client = client ?? http.Client() {
     logger = this;
   }
-  @visibleForTesting
-  http.Client client;
+  final http.Client client;
```

Comment:
Is there a specific reason we're exposing the internal `client`? If not, it'd be better to leave it visible for testing.

```suggestion

  @visibleForTesting
  final http.Client client;
```

Reply:
I couldn't remember a reason for this, so I restored these lines. Thanks!